resources/replace_or_add.rb

```diff
@@ -16,30 +17,30 @@
   add_line = chomp_eol(new_resource.line)
   found = false
   regex = new_resource.pattern.is_a?(String) ? /#{new_resource.pattern}/ : new_resource.pattern
-  new = []
+  new_content = []
   current = target_current_lines
 
   # replace
   current.each do |line|
-    line = line.dup
     if line =~ regex || line == add_line
-      found = true
+      next if found && new_resource.remove_duplicates
       line = add_line
+      found = true
     end
-    new << line
+    new_content << line.dup
```

Comment:
Changing the variable name to new_content for this resource makes it the only one of the set that doesn't use new. I'll probably change it back to new.

Reply:
I wanted to reply before you merge: my coworker recommended to change it as 'nee' is a reserved keyword. If you'd like I can make another PR to rename the variable for all the other resources